### ISSUE TITLE
MD024/no-duplicate-heading to flag errors only when there are duplicate titles under the same top level

### ISSUE THREAD

**Issue by hongbo-miao** [DavidAnson/markdownlint-cli2#242]
After upgrading to markdownlint-cli2 to v0.11.0, I got [error](https://github.com/hongbo-miao/hongbomiao.com/actions/runs/6928343506/job/18843962127?pr=12787):

```shell
README.md:594 MD024/no-duplicate-heading Multiple headings with the same content [Context: "## Simulation"]
README.md:614 MD024/no-duplicate-heading Multiple headings with the same content [Context: "## Embedded, IoT, Hardware"]
README.md:642 MD024/no-duplicate-heading Multiple headings with the same content [Context: "## Computer Vision"]
README.md:702 MD024/no-duplicate-heading Multiple headings with the same content [Context: "## Cloud Native"]
README.md:756 MD024/no-duplicate-heading Multiple headings with the same content [Context: "## Testing"]
README.md:762 MD024/no-duplicate-heading Multiple headings with the same content [Context: "## Ops"]
README.md:788 MD024/no-duplicate-heading Multiple headings with the same content [Context: "## Web"]
README.md:802 MD024/no-duplicate-heading Multiple headings with the same content [Context: "### Security"]
```

However, take `## Simulation` for example, it is under multiple first level header.
```md
# Tech Stack
## Simulation
xxx

# Highlights
## Simulation
yyy
```

Also, they have different contents.

I'm wondering if the "MD024/no-duplicate-heading" linter should only throw an error when there's a duplicate title under the same top level. Would considering that scenario as the standard for duplication be better? Thanks! ☺️

**Comment by DavidAnson** [comment]
This scenario is supported today, please have a look at the documentation: https://github.com/DavidAnson/markdownlint/blob/main/doc/md024.md

**Comment by hongbo-miao** [comment]
Thanks @DavidAnson ! ☀️
I should have checked the document first.
